torch/utils/data/dataloader.py

```diff
@@ -48,8 +57,40 @@ def _worker_loop(dataset, index_queue, data_queue, collate_fn, seed, init_fn, wo
     if init_fn is not None:
         init_fn(worker_id)
 
+    manager_handle = None
+    if IS_WINDOWS:
+        kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+        kernel32.OpenProcess.argtypes = (DWORD, BOOL, DWORD)
+        kernel32.OpenProcess.restype = HANDLE
+        kernel32.WaitForSingleObject.argtypes = (HANDLE, DWORD)
+        kernel32.WaitForSingleObject.restype = DWORD
+
+        # Value obtained from https://msdn.microsoft.com/en-us/library/ms684880.aspx
+        SYNCHRONIZE = 0x00100000
+        manager_handle = kernel32.OpenProcess(SYNCHRONIZE, 0, manager_pid)
+
+        if not manager_handle:
+            raise ctypes.WinError(ctypes.get_last_error())
```

Comment:
It would be nice if we could move this platform specific code outside of `_worker_loop`. Perhaps we could do something like a wrapper class?
```python
if IS_WINDOWS:
    class ManagerWatchdog(object):
        def __init__(self, manager_pid):
            ...
        def is_alive(self):
            return ...
else:
    class ManagerWatchdog(object):
        ... # corresponding impl for linux, just uses `getppid`
```
Then the only modification to this function is that you instantiate the watchdog object and call `is_alive` when the timeout happens